src/com/google/javascript/jscomp/Es6RewriteBlockScopedDeclaration.java

```diff
@@ -290,6 +290,9 @@ private void transformLoopClosure() {
           Node initializer = loopNode.getFirstChild();
           loopNode.replaceChild(initializer, IR.empty());
           if (!initializer.isEmpty()) {
+            if (!initializer.isExprResult()) {
```

Comment:
Not that it matters but I think this isn't quite right. We'll end up with an EXPR_RESULT containing a VAR which is strange.